metricbeat/helper/server/tcp/tcp.go

```diff
@@ -96,22 +98,45 @@ func (g *TcpServer) watchMetrics() {
 			logp.Err("Unable to accept connection due to error: %v", err)
 			continue
 		}
-		defer func() {
-			if conn != nil {
-				conn.Close()
-			}
-		}()
 
-		length, err := conn.Read(buffer)
+		go g.handle(conn)
```

Comment:
In the original PR, this statement is wrapped in an `if conn != nil`. Is that needed here too?